src/fluree/db/time_travel.cljc

```diff
@@ -143,6 +145,29 @@
                         {:status 400
                          :error  :db/invalid-time}))))))
 
+(defn datetime->t
+  "Takes an ISO-8601 datetime string and returns a core.async channel with the
+  latest 't' value that is not more recent than that datetime."
+  [db datetime]
+  (go-try
+    (log/debug "datetime->t db:" (pr-str db))
+    (let [epoch-datetime (util/str->epoch-ms datetime)
+          flakes (some-> db
+                         dbproto/-rootdb
+                         (query-range/index-range :post
+                                                  > [const/$_commit:time 0]
+                                                  < [const/$_commit:time epoch-datetime])
+                         <?)]
+      (log/debug "datetime->t index-range:" (pr-str flakes)
+                 #_(keep #(when (some (fn [v] (= v const/$_commit:time)) %)
+                            %) flakes))
+      (when (empty? flakes)
+        (throw (ex-info (str "There is no data as of " datetime)
+                        {:status 400, :error :db/invalid-query})))
+      (let [flake (apply max-key flake/s flakes)]
```

Comment:
The flakes will be ordered already, so you should be able to do `(peek flakes)` as opposed to applying a function over them.